ccl/backupccl: TestBackupRestoreJobTagAndLabel failed

ccl/backupccl.TestBackupRestoreJobTagAndLabel [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12584796?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12584796?buildTab=artifacts#/) on release-23.2 @ [f34238ce250f30efe53ebf5967dffa6cbad7ad26](https://github.com/cockroachdb/cockroach/commits/f34238ce250f30efe53ebf5967dffa6cbad7ad26):


```
=== RUN   TestBackupRestoreJobTagAndLabel
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/95e138e66d69292427dfb9528cf06d04/logTestBackupRestoreJobTagAndLabel320978944
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:165: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    backup_test.go:293: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/backup_test.go:293
        	Error:      	Should be true
        	Test:       	TestBackupRestoreJobTagAndLabel
    testutils.go:289: no Invalid Descriptors
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/95e138e66d69292427dfb9528cf06d04/logTestBackupRestoreJobTagAndLabel320978944
--- FAIL: TestBackupRestoreJobTagAndLabel (29.57s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #113653 ccl/backupccl: TestBackupRestoreJobTagAndLabel failed [C-test-failure O-remote-execution O-robot T-disaster-recovery branch-master release-blocker]
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupRestoreJobTagAndLabel.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-33358

--- a/pkg/ccl/backupccl/BUILD.bazel
+++ b/pkg/ccl/backupccl/BUILD.bazel
@@ -330,7 +330,6 @@ go_test(
         "@com_github_cockroachdb_datadriven//:datadriven",
         "@com_github_cockroachdb_errors//:errors",
         "@com_github_cockroachdb_errors//oserror",
-        "@com_github_cockroachdb_logtags//:logtags",
         "@com_github_cockroachdb_pebble//sstable",
         "@com_github_cockroachdb_pebble//vfs",
         "@com_github_cockroachdb_redact//:redact",

--- a/pkg/ccl/backupccl/backup_test.go
+++ b/pkg/ccl/backupccl/backup_test.go
@@ -25,7 +25,6 @@ import (
 	"path/filepath"
 	"reflect"
 	"regexp"
-	"runtime/pprof"
 	"strconv"
 	"strings"
 	"sync/atomic"
@@ -109,7 +108,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/cockroachdb/errors"
 	"github.com/cockroachdb/errors/oserror"
-	"github.com/cockroachdb/logtags"
 	"github.com/cockroachdb/redact"
 	pgx "github.com/jackc/pgx/v4"
 	"github.com/kr/pretty"
@@ -225,72 +223,6 @@ func TestBackupRestoreMultiNodeRemote(t *testing.T) {
 	remoteFoo := "nodelocal://2/foo"
 
 	backupAndRestore(ctx, t, tc, []string{remoteFoo}, []string{localFoo}, numAccounts, nil)
-}
-
-// TestBackupRestoreJobTagAndLabel runs a backup and restore and verifies that
-// the flows are executed remotely with a job log tag and a job pprof label.
-func TestBackupRestoreJobTagAndLabel(t *testing.T) {
-	defer leaktest.AfterTest(t)()
-	defer log.Scope(t).Close(t)
-	skip.UnderRaceWithIssue(t, 107336)
-
-	const numAccounts = 1000
-	ctx := context.Background()
-	getJobTag := func(ctx context.Context) string {
-		tags := logtags.FromContext(ctx)
-		if tags != nil {
-			for _, tag := range tags.Get() {
-				if tag.Key() == "job" {
-					return tag.ValueStr()
-				}
-			}
-		}
-		return ""
-	}
-	found := false
-	var mu syncutil.Mutex
-	// backupRestoreTestSetupWithParams() writes some data and then, within
-	// workloadsql.Setup() it splits and scatters the ranges. When using 3 nodes
-	// scatter means we only move leases which is usually enough. During stress or
-	// race we see that the leases may not be scattered because the other 2
-	// replicas are not yet initialized. In those cases the leases all stay on
-	// node 1 and therefore the flows run locally and the test fails (because
-	// 'found' stays false). The simple solution here is to use 4 nodes where
-	// scatter moves replicas in addition to leases.
-	numNodes := 4
-	tc, _, _, cleanupFn := backupRestoreTestSetupWithParams(t, numNodes, numAccounts, InitManualReplication,
-		base.TestClusterArgs{
-			ServerArgs: base.TestServerArgs{
-				Knobs: base.TestingKnobs{
-					DistSQL: &execinfra.TestingKnobs{
-						SetupFlowCb: func(ctx context.Context, _ base.SQLInstanceID, _ *execinfrapb.SetupFlowRequest) error {
-							mu.Lock()
-							defer mu.Unlock()
-							tag := getJobTag(ctx)
-							label, ok := pprof.Label(ctx, "job")
-							if tag == "" && !ok {
-								// Skip, it's not a job.
-								return nil
-							}
-							if tag == "" || !ok || tag != label {
-								log.Fatalf(ctx, "the job tag should exist and match the pprof label: tag=%s label=%s ctx=%s",
-									tag, label, ctx)
-							}
-							found = true
-							return nil
-						},
-					},
-				},
-			},
-		},
-	)
-	defer cleanupFn()
-
-	backupAndRestore(ctx, t, tc, []string{localFoo}, []string{localFoo}, numAccounts, nil)
-
-	mu.Lock()
-	defer mu.Unlock()
-	require.True(t, found)
 }
 
 func findSST(t *testing.T, location string) string {

